spyder_kernels/customize/code_runner.py

```diff
@@ -251,6 +309,41 @@ def debug_exec(code, glob, loc):
             # Enter recursive debugger
             yield debug_exec
 
+    @contextmanager
+    def _profile_exec(self):
+        """Get an exec function for profiling."""
+        with tempfile.TemporaryDirectory() as tempdir:
```

Comment:
This directory needs to be `xdg_data_home` on Linux, as you did before for temp fault handler files. Otherwise, we'd be unnecessarily polluting users' `/tmp` directory.

Reply:
The file is deleted at the end of the execution so it shouldn't polite the /tmp directory

Reply:
Sure, but until the execution has finished the file would be available in `/tmp`. And if the execution takes a long time, then it'd end up polluting it. Another thing to consider is that you profile in as many consoles as you want.

Since `TemporaryDirectory` accepts a `dir` parameter, then please `xdg_data_home` to it.